x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -678,6 +685,47 @@ private List<Attribute> resolveUsingColumns(List<Attribute> cols, List<Attribute
             return resolved;
         }
 
+        private LogicalPlan resolveInsist(Insist insist, List<Attribute> childrenOutput, IndexResolution indexResolution) {
+            Attribute resolvedCol = maybeResolveAttribute(insist.getInsistIdentifier(), childrenOutput);
+            // Field isn't mapped anywhere.
+            if (resolvedCol instanceof UnresolvedAttribute) {
+                return pushdownInsist(insist, attrs -> attrs.add(insistKeyword(insist)));
+            }
+
+            String name = resolvedCol.name();
+            // Field is partially unmapped.
+            if (resolvedCol instanceof FieldAttribute f && indexResolution.get().partiallyUnmappedFields().contains(name)) {
+                return pushdownInsist(insist, attrs -> {
+                    var index = CollectionUtils.findFirstIndex(attrs, e -> e.name().equals(name)).getAsInt();
```

Comment:
This blows up if the name is not found. Instead perform a check and place the error message into the unresolvedAttribute to b picked up by the verifier.